packages/sdk-metrics/src/InstrumentDescriptor.ts

```diff
@@ -71,10 +72,16 @@ export function isDescriptorCompatibleWith(
   descriptor: InstrumentDescriptor,
   otherDescriptor: InstrumentDescriptor
 ) {
+  // Names are case-insensitive strings.
   return (
-    descriptor.name === otherDescriptor.name &&
+    equalsCaseInsensitive(descriptor.name, otherDescriptor.name) &&
     descriptor.unit === otherDescriptor.unit &&
     descriptor.type === otherDescriptor.type &&
     descriptor.valueType === otherDescriptor.valueType
   );
 }
+
+const NAME_REGEXP = /^[a-z][a-z0-9_.-]{0,62}$/i;
```

Comment:
Is always good to include a comment with the description of the regex expression for easier reading
 // ASCII string with a length no greater than 63 characters

Reply:
Names were increased to <=255 character length https://github.com/open-telemetry/opentelemetry-specification/pull/3648

```suggestion
// ASCII string with a length no greater than 255 characters
const NAME_REGEXP = /^[a-z][a-z0-9_.-]{0,255}$/i;
```